crates/re_viewer/src/app_blueprint.rs

```diff
@@ -13,63 +13,111 @@ pub const TIME_PANEL_PATH: &str = "time_panel";
-            blueprint_panel_state: if screen_size.x > 750.0 {
-                PanelState::Expanded
-            } else {
-                PanelState::Collapsed
-            },
-            selection_panel_state: if screen_size.x > 1000.0 {
-                PanelState::Expanded
-            } else {
-                PanelState::Collapsed
-            },
-            time_panel_state: if screen_size.y > 600.0 {
-                PanelState::Expanded
-            } else {
-                PanelState::Collapsed
+            panel_states: PanelStates {
+                top: PanelState::Expanded,
+                blueprint: if screen_size.x > 750.0 {
+                    PanelState::Expanded
+                } else {
+                    PanelState::Collapsed
+                },
+                selection: if screen_size.x > 1000.0 {
+                    PanelState::Expanded
+                } else {
+                    PanelState::Collapsed
+                },
+                time: if screen_size.y > 600.0 {
+                    PanelState::Expanded
+                } else {
+                    PanelState::Collapsed
+                },
             },
+            overrides,
         };
 
         if let Some(blueprint_db) = blueprint_db {
             if let Some(state) = load_panel_state(&TOP_PANEL_PATH.into(), blueprint_db, query) {
-                ret.top_panel_state = state;
+                ret.panel_states.top = state;
             }
             if let Some(state) = load_panel_state(&BLUEPRINT_PANEL_PATH.into(), blueprint_db, query)
             {
-                ret.blueprint_panel_state = state;
+                ret.panel_states.blueprint = state;
             }
             if let Some(state) = load_panel_state(&SELECTION_PANEL_PATH.into(), blueprint_db, query)
             {
-                ret.selection_panel_state = state;
+                ret.panel_states.selection = state;
             }
             if let Some(state) = load_panel_state(&TIME_PANEL_PATH.into(), blueprint_db, query) {
-                ret.time_panel_state = state;
+                ret.panel_states.time = state;
             }
         }
 
         ret
     }
 
+    pub fn top_panel_state(&self) -> PanelState {
+        self.overrides.top.unwrap_or(self.panel_states.top)
+    }
+
+    pub fn blueprint_panel_state(&self) -> PanelState {
+        self.overrides
+            .blueprint
+            .unwrap_or(self.panel_states.blueprint)
+    }
+
+    pub fn selection_panel_state(&self) -> PanelState {
+        self.overrides
+            .selection
+            .unwrap_or(self.panel_states.selection)
+    }
+
+    pub fn time_panel_state(&self) -> PanelState {
+        self.overrides.time.unwrap_or(self.panel_states.time)
+    }
+
+    pub fn toggle_top_panel(&self, command_sender: &CommandSender) {
+        // don't toggle if it is overridden
+        if self.overrides.top.is_some() {
+            return;
+        }
+
```

Comment:
Are we sure about this?

These immutable overrides enforces a form of "kiosk mode", where the end-user isn't allowed to open up the panels. If we are going here, we should selectively remove the toggle buttons and commands, otherwise we will present buttons to the user that don't work.

Alternatively we implement this here by toggling the override, meaning the overrides act as override for the blueprint setting, but are mutable (would require a Mutex). This gives the end-user more power and trust. It would also give an escape hatch for opening these panels for debugging etc.

I think option two is more attractive, unless we have a good user store for "kiosk mode".

Reply:
The primary use case for this is viewer in docs, where we want to force all panels to be disabled no matter what the app blueprint says. But at the same time, if each in-docs viewer can be toggled to fullscreen (coming soon :tm:), then it makes sense to disable the override at that point, so the user can explore the full web viewer.

In that sense, maybe we want a command to toggle the override on/off, so we (and also users) don't have to unset all the panel overrides in JS.